homeassistant/components/shelly/utils.py

```diff
@@ -35,54 +43,76 @@ def get_device_name(device: aioshelly.Device) -> str:
     return device.settings["name"] or device.settings["device"]["hostname"]
 
 
+def get_number_of_channels(device: aioshelly.Device, block: aioshelly.Block) -> int:
+    """Get number of channels for block type."""
+    channels = None
+
+    if block.type == "input":
+        # Shelly Dimmer/1L has two input channels and missing "num_inputs"
+        if device.settings["device"]["type"] in ["SHDM-1", "SHDM-2", "SHSW-L"]:
+            channels = 2
+        else:
+            channels = device.shelly.get("num_inputs")
+    elif block.type == "emeter":
+        channels = device.shelly.get("num_emeters")
+    elif block.type in ["relay", "light"]:
+        channels = device.shelly.get("num_outputs")
+    elif block.type in ["roller", "device"]:
+        channels = 1
+
+    return channels or 1
```

Comment:
Just return as soon as you're done:

```suggestion
            return 2

        return device.shelly.get("num_inputs")
    if block.type == "emeter":
        return device.shelly.get("num_emeters")
    if block.type in ["relay", "light"]:
        return device.shelly.get("num_outputs")
    if block.type in ["roller", "device"]:
        return 1

    return 1
```

Reply:
Offline discussion,  since some devices does not return a value we use 1 instead so we leave the code as before the suggestion.